core/src/main/java/org/infinispan/persistence/spi/CacheLoader.java

```diff
@@ -29,8 +30,24 @@
     *
     * @return the entry, or null if the entry does not exist
     * @throws PersistenceException in case of an error, e.g. communicating with the external storage
+    * @deprecated since 10.0 please implement {{@link #get(Object)}} instead
     */
-   MarshalledEntry<K, V> load(Object key);
+   @Deprecated
+   default MarshalledEntry<K, V> load(Object key) {
+      return null;
+   }
+
+   /**
+    * Fetches an entry from the storage. If a {@link MarshallableEntry} needs to be created here, {@link
+    * InitializationContext#getMarshallableEntryFactory()} ()} and {@link
+    * InitializationContext#getByteBufferFactory()} should be used.
+    *
+    * @return the entry, or null if the entry does not exist
+    * @throws PersistenceException in case of an error, e.g. communicating with the external storage
+    */
+   default MarshallableEntry<K, V> get(Object key) {
```

Comment:
I was thinking stores could still return `MarshalledEntry`, because the language allows covariant return types. But it requires recompilation, otherwise there's no bridge method returning a `MarshallableEntry` in the store implementation...

I'm still not sold on `get()` though, it's weird to have an interface `CacheLoader` where the without a proper `load` method. I quickly gave up on generating the bytecode directly to avoid the compiler's silly restriction on methods that only differ in their return type, so now I'm trying to come up with something that includes `load` in the name, like `loadEntry(key)` or `load(key, includeExpired)`. WDYT @wburns @tristantarrant?

Reply:
> I was thinking stores could still return `MarshalledEntry`

So that's a problem in itself, as it means we can't properly deprecate `MarshalledEntry` as there will be no new method for the user to migrate to. So at what point can we remove it without breaking compatibility?